### PR TITLE
Clean up delphi_utils 

### LINKED ISSUE #1973: Clean up delphi_utils
`delphi_utils` is a big hodge-podge of a package with a lot of dependencies, thus it has a relatively large net footprint and takes a while to install.  Lets see if we can lessen that.

Move [packages in setup.py](https://github.com/cmu-delphi/covidcast-indicators/blob/a6ea00355d79a47cc09db8090f6e5121c718de0f/_delphi_utils_python/setup.py#L8-L26) that are unnecessary for a downstream user but still used in test/lint/ci, like: `darker`, `freezegun`, `mock`, `moto`, `pydocstyle`, `pylint`, `pytest-cov`, `pytest`, `requests-mock`.  They can be put into something like a "`requirements.test.txt`" file that gets installed via [the CI](https://github.com/cmu-delphi/covidcast-indicators/blob/a6ea00355d79a47cc09db8090f6e5121c718de0f/.github/workflows/python-ci.yml#L53-L70) and/or [Makefile](https://github.com/cmu-delphi/covidcast-indicators/blob/a6ea00355d79a47cc09db8090f6e5121c718de0f/_delphi_utils_python/Makefile#L11-L25).  

Remove from `setup.py` any unused packages, as `gitpython` appears to be.

The `covidcast` package has its own [wacky imports](https://github.com/cmu-delphi/covidcast/blob/main/Python-packages/covidcast-py/setup.py#L23-L33), which includes a bunch of imaging/graphics and geospatial libraries.  It is only used in the `delphi_utils.validator` module for pulling data from the API and can be replaced with the lighter-weight [`delphi_epidata` client](https://pypi.org/project/delphi-epidata/) instead.  Doing that is not quite as easy as moving requirements entries around, and it is covered in in #1972.

### PR DESCRIPTION
**Pull request by aysim319**
### Description
migrating setup.py to pyproject.toml and cleaning up dependencies

### Changelog
Itemize code/test/documentation changes and files added/removed.
- added pyproject.toml

### Fixes 
- Fixes #1973

### CODE REVIEW

File: _delphi_utils_python/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    "importlib_resources>=1.3",
+    "numpy",
+    "pandas>=1.1.0",
+    "slackclient",
+    "structlog",
+    "xlrd",
+]
+
+[project.urls]
+Homepage = "https://github.com/cmu-delphi/"
```
Comment (dshemetov) [inline _delphi_utils_python/pyproject.toml:32]:
```suggestion
Homepage = "https://github.com/cmu-delphi/covidcast-indicators"
```

**Review by dshemetov — CHANGES_REQUESTED** [review]
On the right track, thanks! Can you please also
- [x] adjust the [bumpversion.cfg file](https://github.com/cmu-delphi/covidcast-indicators/blob/84d059751b646c0075f1a384741f2c1d80981269/_delphi_utils_python/.bumpversion.cfg#L5) to replace setup.py with pyproject.toml
- [x] update the release command in the [Makefile](https://github.com/cmu-delphi/covidcast-indicators/blob/84d059751b646c0075f1a384741f2c1d80981269/_delphi_utils_python/Makefile#L31-L33) to do `python -m build --sdist --wheel` 
- [x] and update [this CI line](https://github.com/cmu-delphi/covidcast-indicators/blob/84d059751b646c0075f1a384741f2c1d80981269/.github/workflows/publish-release.yml#L82) to `pip install build pylint pytest pydocstyle wheel twine`
- [x] update this [install line](https://github.com/cmu-delphi/covidcast-indicators/blob/84d059751b646c0075f1a384741f2c1d80981269/_delphi_utils_python/Makefile#L9) to `pip install -e .[dev]`
- [x] remove `setup.py`
- [x] update the unit test install line in `DEVELOP.md` to `pip install -e .[dev]`

Would also appreciate if you can sanity check some of these commands locally.

File: _delphi_utils_python/Makefile [hunk 2]
```diff
@@ -6,12 +6,11 @@ venv:
...
 	. env/bin/activate; \
 	pip install wheel ; \
-	pip install -e .
+	pip install -e '.[dev]'
 
 install-ci: venv
 	. env/bin/activate; \
-	pip install wheel ; \
-	pip install .
+	pip install 'build[virtualenv]' pylint pytest pydocstyle wheel twine
```
Comment (dshemetov) [inline _delphi_utils_python/Makefile:13]:
```suggestion
	pip install 'build[virtualenv]' pylint pytest pydocstyle wheel twine
	pip install '.[dev]'
```

**Comment by aysim319** [comment]
release diff: 

STR:
*  run `make  release` in this branch
    - rename dist to dist_new 
*  run `make  release` in main
* unzip both tar
* in Mac ` diff -rq dist/delphi_utils-xxxx dist_new/delphi_utils-yyyy
    - the versioning might be off 

result

```
Files dist/delphi_utils-0.3.24/.bumpversion.cfg and dist_new/delphi_utils-0.3.23/.bumpversion.cfg differ
Files dist/delphi_utils-0.3.24/DEVELOP.md and dist_new/delphi_utils-0.3.23/DEVELOP.md differ
-> command changed reflected in PR
Files dist/delphi_utils-0.3.24/Makefile and dist_new/delphi_utils-0.3.23/Makefile differ
-> commands change reflected in the PR
Files dist/delphi_utils-0.3.24/PKG-INFO and dist_new/delphi_utils-0.3.23/PKG-INFO differ
-> package info changed reflected in PR
Files dist/delphi_utils-0.3.24/delphi_utils/.DS_Store and dist_new/delphi_utils-0.3.23/delphi_utils/.DS_Store differ
Files dist/delphi_utils-0.3.24/delphi_utils/data/.DS_Store and dist_new/delphi_utils-0.3.23/delphi_utils/data/.DS_Store differ
Files dist/delphi_utils-0.3.24/delphi_utils.egg-info/PKG-INFO and dist_new/delphi_utils-0.3.23/delphi_utils.egg-info/PKG-INFO differ
Files dist/delphi_utils-0.3.24/delphi_utils.egg-info/SOURCES.txt and dist_new/delphi_utils-0.3.23/delphi_utils.egg-info/SOURCES.txt differ
Files dist/delphi_utils-0.3.24/delphi_utils.egg-info/requires.txt and dist_new/delphi_utils-0.3.23/delphi_utils.egg-info/requires.txt differ
Only in dist_new/delphi_utils-0.3.23: pyproject.toml
Only in dist/delphi_utils-0.3.24: setup.py
```

File: _delphi_utils_python/pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,60 @@
+[build-system]
+requires = ["setuptools", "setuptools-scm>=8.0"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "delphi-utils"
+version = "0.3.23"
```
Comment (dshemetov) [inline _delphi_utils_python/pyproject.toml:7]:
```suggestion
version = "0.3.24"
```

**Review by dshemetov — APPROVED** [review]
This looks good to me. Tests pass, so we can be confident our indicators will continue to work. Amaris tested the release command as well and the artifacts didn't have any unexpected changes. I double checked that all the dependencies have been moved from setup.py over to pyproject.toml.

File: _delphi_utils_python/pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+]
+dependencies = [
+    "boto3",
+    "covidcast",
+    "cvxpy",
+    "epiweeks",
+    "gitpython",
+    "importlib_resources>=1.3",
+    "numpy",
+    "pandas>=1.1.0",
```
Comment (aysim319) [inline _delphi_utils_python/pyproject.toml:26]: Do we change the version of this to 2.2.2?

Comment (dshemetov) [reply _delphi_utils_python/pyproject.toml:26]: Not yet, our pandas was indirectly pinned to <1.5.3 through our dependence on `covidcast` (more info [here](https://github.com/cmu-delphi/covidcast-indicators/issues/1972#issuecomment-2197688725)) and the change to 2.0.0 brought a number of breaking changes, so we should do the Pandas upgrade slowly.

**Pull request merged**